### PR TITLE
feat(launch): Default to using model-registry project for agent and launch_add

### PR DESCRIPTION
**Pull request by KyleGoyette**
[Fixes WB-11514](https://wandb.atlassian.net/browse/WB-11514)
[Fixes WB-11515](https://wandb.atlassian.net/browse/WB-11515)

Fixes #NNNN

Description
-----------

- allows users to specify the project that a queue resides in when queueing a job
- makes the model-registry project the default project when queueing a job, or for the launch-agent

Testing
-------
How was this PR tested?

Checklist
-------
- [ ] Include reference to internal ticket "Fixes WB-NNNN" and/or GitHub issue "Fixes #NNNN" (if applicable)
- [ ] Ensure PR title compliance with the [conventional commits standards](https://github.com/wandb/wandb/blob/master/CONTRIBUTING.md#conventional-commits)

### CODE REVIEW

File: wandb/cli/cli.py [hunk 1]
```diff
@@ -697,6 +701,11 @@ def _clean():
     default=None,
     help="Set sweep name",
 )
+@click.option(
+    "--project_queue",
```
Comment (gtarpenning) [inline wandb/cli/cli.py:705]: I dislike this being an underscore but I guess we sealed our fate by using `launch_config`. In the launch agent, `--max-jobs` is the param not `--max_jobs`

File: wandb/cli/cli.py [hunk 2]
```diff
@@ -1126,6 +1137,13 @@ def _parse_settings(settings):
     default=None,
     help="Name of a remote repository. Will be used to push a built image to.",
 )
+# TODO: this is only included for back compat. But we should remove this in the future
+@click.option(
+    "--project-queue",
```
Comment (gtarpenning) [inline wandb/cli/cli.py:1142]: !! so gross see above

File: wandb/sdk/internal/internal_api.py [hunk 3]
```diff
@@ -1080,23 +1080,25 @@ def push_to_run_queue_by_name(
 
         return result
 
-    @normalize_exceptions
+    # @normalize_exceptions
```
Comment (gtarpenning) [inline wandb/sdk/internal/internal_api.py:1083]: Interesting, stack trace implications?

**Review by gtarpenning — APPROVED** [review]
assuming tests pass I think this makes sense. I'm curious what happens to run queue items that are currently queued, and then get dequeued after the change. but lgtm

File: tests/unit_tests/tests_launch/test_launch_add.py [hunk 4]
```diff
@@ -99,10 +104,10 @@ def test_launch_add_delete_queued_run(
 
 
 # TODO(gst): Identify root cause of (threaded?) artifact creation error
-@pytest.mark.xfail(
-    strict=False,
-    reason="Non-deterministic, 1-2 can fail but all 4 would suggest regression.",
-)
+# @pytest.mark.xfail(
```
[the remark this answers is not in the document]
Comment (KyleGoyette) [reply tests/unit_tests/tests_launch/test_launch_add.py:107]: oh, nope this was for testing, I'll remove

File: wandb/sdk/internal/internal_api.py (hunk 3 above)
Comment (KyleGoyette) [reply wandb/sdk/internal/internal_api.py:1083]: just for testing

File: wandb/cli/cli.py (hunk 1 above)
Comment (KyleGoyette) [reply wandb/cli/cli.py:705]: Im going all in on hyphens

File: wandb/sdk/launch/launch.py [hunk 5]
```diff
@@ -87,6 +94,11 @@ def resolve_agent_config(
         )
         api = Api(default_settings=settings)
 
+    if user_set_project:
+        wandb.termwarn(
+            "Specifying a project for the launch agent will be deprecated. Please use queues found in the Launch application at https://wandb.ai/launch."
```
Comment (jamie-rasmussen) [inline wandb/sdk/launch/launch.py:99]: "will be deprecated" or "is deprecated"?

**Review by jamie-rasmussen — APPROVED** [review]
It seems like we're leaking the internal implementation to the user a bit, e.g. the output of running launch-agent says, "polling on project model-registry"

**Comment by KyleGoyette** [comment]
Fixed the info leak. Nice catch.

**Review by raubitsj — APPROVED** [review]
reviewed internal_api change. looks good.

**Pull request merged**